src/catalog/src/builtin.rs

```diff
@@ -5947,20 +5947,62 @@ ON mz_internal.mz_sink_status_history (sink_id)",
     is_retained_metrics_object: false,
 };
 
+pub const MZ_SOURCE_STATISTICS: BuiltinView = BuiltinView {
+    name: "mz_source_statistics",
+    schema: MZ_INTERNAL_SCHEMA,
+    column_defs: None,
+    // everything but `params`
+    sql: "
+SELECT
+    id,
+    bool_and(snapshot_committed) as snapshot_committed,
+    SUM(messages_received)::uint8 AS messages_received,
```

Comment:
These shouldn't be cast back to their original type. The upgrade to `numeric` is intentional.

```
postgres=# create table n (i bigint);
CREATE TABLE
postgres=# select pg_typeof(sum(i)) from n;
 pg_typeof 
-----------
 numeric
```

Reply:
ah, will do!

Reply:
Chatted on Slack and hadn't realized that the memory pressure is a concern. These are fine to cast back to uint8 assuming the values are guaranteed to fit. Sorry for the alarm.

Reply:
discussed on slack; will keep the cast as i don't expect the sum to be > 2^63 (even if a source on a 6xlarge source produces 400 terabytes in a week on each worker its 2^59)

Reply:
Can you add the rationale as a comment in the code?

Reply:
yes!